### PR TITLE
Add functionality to update user account details

### PR DESCRIPTION
**Pull request by swapnilkamat23**
This PR adds -
1. Update functionality for user account details
2. React hook form for easy state management, handle submit and form validation and user feedback
3. Toast notifications

### CODE REVIEW

File: app/views/screens/AccountScreen.tsx [hunk 1]
```diff
@@ -1,37 +1,117 @@
...
+            toast
+          );
+          asyncStorage.storeDataObject(USER_OBJECT_KEY, user.data);
+        } else {
+          console.log("Error when calling v1/getUser api - ", user.data);
+          showNotificationToast(
+            "User account details failed to save. Please try again.",
+            TOAST_TYPE.FAILURE,
+            toast
+          );
```
Comment (ndedhia1210) [inline app/views/screens/AccountScreen.tsx]: I think you can avoid this service call. 
so if UpdateUserApi succeeds then simply the replace older user object with new user object (basically replacing current value with user entered values).

File: app/views/screens/LoginScreen.tsx [hunk 2]
```diff
@@ -78,28 +79,30 @@ function LoginScreen(props) {
...
             textContentType="password"
-            onChangeText={(text) => {
-              errorMessage && setErrorMessage("");
-              setPassword(text);
+            rules={{
+              required: "Password is required",
+              minLength: {
+                value: 8,
+                message: "Password should be at least 8 characters long",
+              },
```
Comment (ndedhia1210) [inline app/views/screens/LoginScreen.tsx]: Hey, backend have this criteria, can we do the same here? https://github.com/ndedhia1210/smallBusiness/blob/main/app/user/resetUserPassword.js#L29

Password must be alphanumeric length can be from 5-10 (We can change BE to make it 8-12 length but for now can we use this and check for alpha numeric too)

File: app/views/screens/AccountScreen.tsx (hunk 1 above)
Comment (swapnilkamat23) [reply app/views/screens/AccountScreen.tsx]: Done! I was thinking about consistency so had written it that way. Removed it now.

File: app/views/screens/LoginScreen.tsx (hunk 2 above)
Comment (swapnilkamat23) [reply app/views/screens/LoginScreen.tsx]: Done! Required, 5-10 chars, alphanumeric.

**Review by ndedhia1210 — APPROVED** [review]

**Review by vaibhavnagda — APPROVED** [review]
🚀

**Pull request merged**